### ISSUE TITLE
Upgrade to SL5

### ISSUE THREAD

**Issue by crmckenzie** [nbuilder/nbuilder#48]
I'm trying to get everything "current" as of the last 2 years. The SL tests are not yet passing. I may bring in @cmbrown1598 to help as he has recent problem-solving skills in this space.

**Comment by PureKrome** [comment]
A question about this : do we actually _care_ about Silverlight? Honest Q - not trolling.

It's dead. So why bother? There are versions of NBuilder already on nuget.org that are supporting it .. so why spend the effort on something that might have little-to-none ROI ?

My vote: nuke SL support.

**Comment by crmckenzie** [comment]
The last time I talked with @garethdown44  about this (2 years ago?) he was pretty sure he wanted to maintain Silverlight support. However, since Silverlight does seem to be going the way of the quaga _and_ since it is becoming an impediment to modernizing the project, I would be happy to see it go. The best argument I have for it for the time being is that learning how to make this work in the build system will go a long way to preparing NBuilder for .NET Core. On the other hand, it may be more pragmatic to modernize the project around .NET 4 and worry about .NET Core after it's released.

@garethdown44, what are your thoughts?

**Comment by pjmagee** [comment]
Nuke SL and move towards supporting .NET Core

**Comment by PureKrome** [comment]
Also +1 to nuking SL and sticking with modern and dealing with .NET Core after 1.0 RTM's and the initial shit storm dies down or shit gets fixed in .net core post 1.0.

Summary IMO: rebase to 4.x and start dealing with .net core in a few months.

**Comment by garethdown44** [comment]
Thing is, there's at least one banking project I know using it. I was at someone's desk a few weeks ago and noticed it was used in their code base. Lots of projects in the banking space still use SL as there was a big push across the industry a few years ago.

How about keeping the support, only in the sense that you don't actually rip it out, but don't bother with CI. If there turns out to be a problem, someone can submit a pull request.

Basically spend no more time on it but don't forcibly remove support.

Would that work?

**Comment by PureKrome** [comment]
Yeah - that could work.

or ... just remove it from the latest master branch and if the support requires a patch fix, then you can branch of a commit or tag.

Removing it doesn't remove it from nuget.org where existing players can still download it.

**Comment by crmckenzie** [comment]
It could work, but it makes things complicated. I don't like the idea of dropping tests (it makes me feel dirty), but given that some of the test dependencies don't exist on Nuget, I don't see any alternative that lets us modernize the repo in preparation for .NET Core. 

One argument is that if these banking customers have been using NBuilder with no updates or reported issues for 2 years then they're not likely to need further support. They could continue forward with the version of NBuilder they have now and their world would not change. Dropping support from the next and future versions should not affect them.

Alternatively, we could split the repo into two and maintain a separate NBuilder.Silverlight repo & package for compatibility and patch purposes but not new features.

I'm open to other ideas, but at some point we will need to drop support for SL. I don't think it's realistic or even good for the project to maintain support indefinitely. Perhaps this discussion could be clarified if @garethdown44  would be so kind as to state what the criteria should be for dropping support for SL altogether.

**Comment by PureKrome** [comment]
Having two repo's sounds interesting - that's an obvious *line-drawn-in-the-sand* announcements. Downside could be that some people might get confused -- why two repos? huh? wa?

Either has pro/cons - so just make a call and move on.

**Comment by pjmagee** [comment]
I guess a branch also wouldn't fit?

**Comment by crmckenzie** [comment]
Make that 5 years since NBuilder was released. @garethdown44, I really think it's time to let go of Silverlight. If you disagree can you "state what the criteria should be for dropping support for SL" so that we'll know when the time is right?

**Comment by pjmagee** [comment]
We should tag a commit that says "SL is nuked after this point". Make a seperate SL branch in this github repository. Then in the read me, clearly state that silverlight is not supported but fixes or patches can be made to an old "[LINK TO BRANCH]" of NBuilder, that will not go published on NuGet under NBuilder, but maybe under NBuilder.Legacy NuGet (If we wanted to continue to support publishing an old legacy NBuilder) 

But I honestly think, nuke it all. Its open source software, we want to take it in the right direction and ensure the project lives on and takes ground in .NET Core, we can't be holding onto the past. A bank can easily hire a few contractors to modify an old NBuilder commit...if they need to.

**Comment by rolfwessels** [comment]
Hi, Is there any movement on this? I'm a bit fan of the nbuilder and I'm keen to add it back into .net core projects.

**Comment by crmckenzie** [comment]
Hi @rolfwessels ,
My plan is to issue a catch-up release of the cumulative changes and bug fixes that have amassed over the last several years and turn my attention to the DotNetCore version. I have a PR with a DotNetCore implementation, but it will take some work to integrate as a) it was based off of the master branch instead of develop, and b) it creates copies of the source files instead of linking to the originals. 

DotNetCore will be coming to NBuilder. The issue is being tracked here:
https://github.com/nbuilder/nbuilder/issues/55

**Comment by rolfwessels** [comment]
Thanks, looking forward to it. Happy to help if you need anything.
